storage/xtradb/handler/handler0alter.cc

```diff
@@ -201,6 +201,49 @@ innobase_fulltext_exist(
 
 	return(false);
 }
+/*******************************************************************//**
+Determine if ALTER TABLE needs to rebuild the table .
+@param ha_alter_info		the DDL operation
+@param old_table			old table
+@return whether it is necessary to rebuild the table */
+static __attribute__((nonnull, warn_unused_result))
+bool
+innobase_need_rebuild_for_hash_cols(
+/*==================*/
+    Alter_inplace_info*	ha_alter_info,
+    dict_table_t*			old_table)
+{
+	/*If one of the hash indexes is dropped then
+	hash column associated with it needs to be dropped*/
+	for (uint i = 0; i < ha_alter_info->index_drop_count; i++) {
+		const KEY*	key
+			= ha_alter_info->index_drop_buffer[i];
+		dict_index_t*	index
+			= dict_table_get_index_on_name_and_min_id(
+				old_table, key->name);
+
+		if (index && index->type & DICT_UNIQUE_HASH) {
+			ha_alter_info->handler_flags |= Alter_inplace_info::DROP_COLUMN;
```

Comment:
You mark alter info that there is also need to drop column, but the dropped column is not added anywhere, this could be correct, but needs to be careful if original alter table already contained also drop column part.